cg/apps/sequencing_metrics_parser/parsers/bcl2fastq.py

```diff
@@ -104,3 +109,73 @@ def get_bcl2fastq_stats_paths(demultiplex_result_directory: Path) -> List[Path]:
         )
 
     return stats_json_paths
+
+
+def create_default_sample_lane_metrics(
+    flow_cell_name: str, lane: int, sample_id: str
+) -> SampleLaneMetrics:
+    return SampleLaneMetrics(
+        flow_cell_name=flow_cell_name,
+        lane=lane,
+        sample_id=sample_id,
+        total_reads=0,
+        total_yield=0,
+        total_yield_q30=0,
+        total_quality_score=0,
+    )
+
+
+def sum_quality_scores(read_metrics: Iterable[ReadMetric]) -> int:
+    return sum([read_metric.quality_score_sum for read_metric in read_metrics])
+
+
+def sum_q30_yields(read_metrics: Iterable[ReadMetric]) -> int:
+    return sum([read_metric.yield_q30 for read_metric in read_metrics])
+
+
+def update_lane_metrics_with_undetermined_data(
+    sample_lane_metric: SampleLaneMetrics, undetermined_data: Undetermined
+):
+    reads: int = undetermined_data.number_reads
+    sample_lane_metric.total_reads += reads * 2
+
+    yield_: int = undetermined_data.yield_
+    sample_lane_metric.total_yield += yield_
+
+    read_metrics: List[ReadMetric] = undetermined_data.read_metrics
+    sample_lane_metric.total_yield_q30 += sum_q30_yields(read_metrics)
+    sample_lane_metric.total_quality_score += sum_quality_scores(read_metrics)
+
+
+def combine_undetermined_lane_metrics_from_tiles(
+    tile_metrics: List[SampleLaneTileMetrics],
+) -> List[SampleLaneMetrics]:
+    """Aggregate the undetermined tile metrics per lane."""
+    sample_lane_metrics = {}
+
+    for tile_metric in tile_metrics:
+        for conversion_result in tile_metric.conversion_results:
+            if conversion_result.undetermined is None:
+                continue
+
+            lane: int = conversion_result.lane_number
+            flow_cell_name: str = tile_metric.flow_cell_name
+
+            if lane not in sample_lane_metrics:
+                sample_lane_metrics[lane] = create_default_sample_lane_metrics(
+                    flow_cell_name=flow_cell_name,
+                    lane=lane,
+                    sample_id="undetermined",
```

Comment:
The sample id will be possible to assign for lanes containing a single sample. I think that logic should be elsewhere. Let me know if you have any ideas

Reply:
We should assign the sample internal id as soon as possible in the flow I suppose. Where do we get it from? Is it the sample sheet and look if there is a sample on a single lane?

Reply:
Exactly, we get them from the sample sheet 👍

Reply:
I think we should set the sample internal id here then already, what do others think?

Reply:
Yes, I'm leaning towards this as well

Reply:
Done below in `get_non_pooled_undetermined_metrics_and_assign_sample_ids` instead.